### PR TITLE
Fixes #7935 - Review HTTP/2 error handling

### LINKED ISSUE #7935: Review HTTP/2 error handling (CVE-2022-2048)
**Jetty version(s)**
9.4.x+

**Description**
Review HTTP/2 error handling, in particular during request processing.

**Fixes Security Advisory**
https://github.com/eclipse/jetty.project/security/advisories/GHSA-wgmr-mf83-7x4j
CVE-2022-2048

### PR DESCRIPTION
**Pull request by sbordet**
Now returning error handling code as a Runnable.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

File: jetty-http2/http2-server/src/test/java/org/eclipse/jetty/http2/server/BadURITest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            }
+        });
+
+        // Remove existing ErrorHandlers.
+        while (true)
+        {
+            ErrorHandler errorHandler = server.getBean(ErrorHandler.class);
+            if (errorHandler == null)
+                break;
+            server.removeBean(errorHandler);
```
Comment (joakime) [inline jetty-http2/http2-server/src/test/java/org/eclipse/jetty/http2/server/BadURITest.java:87]: Couldn't you use?

``` java
for (ErrorHandler errorHandler: server.getBeans(ErrorHandler.class))
{
    server.removeBean(errorHandler);
}
```

Comment (sbordet) [reply jetty-http2/http2-server/src/test/java/org/eclipse/jetty/http2/server/BadURITest.java:87]: Done.

**Review by joakime — APPROVED** [review]

**Review by gregw — APPROVED** [review]
I'm re running the CI to see if the failure was a flake.
Other than that... LGTM

**Review by lorban — APPROVED** [review]

**Comment by sbordet** [comment]
Merged to `jetty-12.0.x` on 11 May 2022, commit b705091afb9158b39590fa36b9b84795524a8f0b.
Relevant changes have been moved from Jetty 11's `HttpChannelOverHTTP2` to Jetty 12's `HttpStreamOverHTTP2`.
Similarly for `HttpStreamOverHTTP3`.

**Pull request merged**